src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -5173,6 +5175,89 @@ namespace HeatBalanceManager {
+                                          "\" no corresponding surface (ref BuildingSurface:Detailed) has been found in the input file.");
+                    ErrorsFound = true;
+                } else {
+                    if (state.dataSurface->Surface(SurfNum).Class == DataSurfaces::SurfaceClass::Window &&
+                        state.dataSurface->Surface(SurfNum).ExtBoundCond == DataSurfaces::ExternalEnvironment) {
+                        int ConstrNum = state.dataSurface->Surface(SurfNum).Construction;
+                        int MaterNum =
+                            state.dataConstruction->Construct(ConstrNum).LayerPoint(state.dataConstruction->Construct(ConstrNum).TotLayers);
+                        bool withNoncompatibleShades =
+                            (state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::Shade ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::WindowBlind ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::Screen ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::GlassEquivalentLayer ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::GapEquivalentLayer ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::ShadeEquivalentLayer ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::DrapeEquivalentLayer ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::ScreenEquivalentLayer ||
+                             state.dataMaterial->Material(MaterNum).Group == DataHeatBalance::MaterialGroup::BlindEquivalentLayer ||
+                             state.dataSurface->Surface(SurfNum).HasShadeControl);
+                        if (withNoncompatibleShades) {
+                            ShowSevereError(state,
+                                            "Non-compatible shades defined alongside SurfaceProperty:IncidentSolarMultiplier for the same window");
+                            ErrorsFound = true;
+                        } else {
+                            state.dataSurface->Surface(SurfNum).hasIncSolMultiplier = true;
+                            state.dataSurface->SurfIncSolMultiplier(SurfNum).Name = state.dataIPShortCut->cAlphaArgs(1);
+                            state.dataSurface->SurfIncSolMultiplier(SurfNum).SurfaceIdx = SurfNum;
+                            state.dataSurface->SurfIncSolMultiplier(SurfNum).Scaler = state.dataIPShortCut->rNumericArgs(1);
+                            state.dataSurface->SurfIncSolMultiplier(SurfNum).SchedPtr = GetScheduleIndex(state, state.dataIPShortCut->cAlphaArgs(2));
```

Comment:
Do you not need to check whether the schedule is valid?

Reply:
if the schedule is not valid, then the "scaler" field will be used.
```
        if (state.dataSurface->SurfIncSolMultiplier(SurfNum).SchedPtr > 0) {
            return GetCurrentScheduleValue(state, state.dataSurface->SurfIncSolMultiplier(SurfNum).SchedPtr) *
                   state.dataSurface->SurfIncSolMultiplier(SurfNum).Scaler;
        } else {
            return state.dataSurface->SurfIncSolMultiplier(SurfNum).Scaler;
        }
```

Reply:
I see. But if that is what the user intends, then they would probably leave the schedule field blank. If they give a bad schedule name, they probably meant to use a schedule but made a mistake. No?

Reply:
Bad schedule name would be a problem. I'll add in a check on that.